src/freertos_drivers/common/Device.cxx

```diff
@@ -144,6 +144,7 @@ int Device::close(struct _reent *reent, int fd)
         // stdin, stdout, and stderr never get closed
         return 0;
     }
+    files[fd].inshdn = true;
```

Comment:
Question: What if f->dev->close(f) returns an error? Should we reset the inshdn back to false? Otherwise, I don't think we can ever free the fd because we cannot make another close() call later.

Reply:
Good catch, fixed.